DynamicPoolAllocator.hpp

```diff
@@ -151,33 +193,113 @@ class DynamicPoolAllocator
     }
   }
 
+  void freeOriginalAllocationBlocks() {
```

Comment:
You might want to rename this method also, since `originalAllocation` is gone. Could just be `freeAllocatedBlocks`.